pygen/pygen_src/riscv_instr_gen_config.py

```diff
@@ -9,26 +9,260 @@
+
+        # Self.s_mem_region = [] # TODO
+
+        self.kernel_stack_len = 4000
+        self.kernel_program_instr_cnt = 400
+        # list of main implemented CSRs
         self.invalid_priv_mode_csrs = []
-        self.init_privileged_mode = "MACHINE_MODE"
+        self.num_of_sub_program = 5
+        self.instr_cnt = 200
+        self.num_of_tests = 1
+        self.no_data_page = 0
+        self.no_branch_jump = 0
+        self.no_load_store = 0
+        self.no_csr_instr = 0
         self.no_ebreak = 1
         self.no_dret = 1
-        self.no_csr_instr = 1
+        self.no_fence = 1
         self.no_wfi = 1
+        self.enable_unaligned_load_store = 0
+        self.illegal_instr_ratio = 0
+        self.hint_instr_ratio = 0
+        self.num_of_harts = rcs.NUM_HARTS
+        self.fix_sp = 0
+        self.use_push_data_section = 0
+        self.boot_mode_opts = ""
+        self.enable_page_table_exception = 0
+        self.no_directed_instr = 0
+        self.asm_test_suffix = ""
+        self.enable_interrupt = 0
+        self.enable_nested_interrupt = 0
+        self.enable_timer_irq = 0
+        self.bare_program_mode = 0
+        self.enable_illegal_csr_instruction = 0
+        self.enable_access_invalid_csr_level = 0
+        self.enable_misaligned_instr = 0
+        self.enable_dummy_csr_write = 0
+        self.randomize_csr = 0
+        self.allow_sfence_exception = 0
+        self.no_delegation = 1
+        self.force_m_delegation = 0
+        self.force_s_delegation = 0
+        self.support_supervisor_mode = 0
+        self.disable_compressed_instr = 0
+        self.signature_addr = 0xdeadbeef
+        self.require_signature_addr = 0
+        self.gen_debug_section = 0
+        self.enable_ebreak_in_debug_rom = 0
+        self.set_dcsr_ebreak = 0
+        self.num_debug_sub_program = 0
+        self.enable_debug_single_step = 0
+        self.single_step_iterations = 0
+        self.set_mstatus_tw = 0
+        self.set_mstatus_mprv = 0
+        self.min_stack_len_per_program = 10 * (rcs.XLEN / 8)
+        self.max_stack_len_per_program = 16 * (rcs.XLEN / 8)
+        self.max_branch_step = 20
+        self.max_directed_instr_stream_seq = 20
+        self.reserved_regs = []
+        self.enable_floating_point = 0
+        self.enable_vector_extension = 0
+        self.enable_b_extension = 0
+        # Commenting out for now
+        # self.enable_bitmanip_groups = ['ZBB', 'ZBS', 'ZBP', 'ZBE', 'ZBF',
+        # 'ZBC', 'ZBR', 'ZBM', 'ZBT', 'ZB_TMP']
+        self.dist_control_mode = 0
+        self.category_dist = {}
+
+    def check_setting(self):
+        support_64b = 0
+        support_128b = 0
+
+        # list of satp_mode_t from riscv_core_setting.py
+        stp_md_lst = rcs.SATP_MODE
+
+        # list of riscv_instr_group_t with names of riscv_instr_name_t in it.
+        supported_isa_lst = list(map(lambda z: z.name, riscv_instr_group_t))
+
+        # check the valid isa support
+        for x in rcs.supported_isa:
```

Comment:
I am curious about why you do not compare the x with hardcode value? For example:
`x == 'RV64I'`. You map the list `supported_isa_lst` from class `enum`.